File: src/main/java/no/hvl/tk/visual/debugger/server/endpoint/UIConfig.java
package no.hvl.tk.visual.debugger.server.endpoint;

import static no.hvl.tk.visual.debugger.util.DiagramToJSONConverter.mapper;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.intellij.openapi.diagnostic.Logger;

public record UIConfig(Integer savedDebugSteps, boolean coloredDiff) {
  private static final Logger LOGGER = Logger.getInstance(UIConfig.class);

  public String serialize() {
    try {
      return mapper.writeValueAsString(this);
    } catch (final JsonProcessingException e) {
      LOGGER.error(e);
      return "JsonProcessingException";
    }
  }
}

File: src/main/java/no/hvl/tk/visual/debugger/server/endpoint/UIConfig.kt
package no.hvl.tk.visual.debugger.server.endpoint

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.intellij.openapi.diagnostic.Logger

@JvmRecord
data class UIConfig(val savedDebugSteps: Int, val coloredDiff: Boolean) {
    fun serialize(): String {
        val mapper = ObjectMapper()
        try {
            return mapper.writeValueAsString(this)
        } catch (e: JsonProcessingException) {
            LOGGER.error(e)
            return "JsonProcessingException: $e"
        }
    }

    companion object {
        private val LOGGER = Logger.getInstance(
            UIConfig::class.java
        )
    }
}
